ERCS/erc-7246.md

```diff
@@ -0,0 +1,221 @@
+        return (balanceOf(a) - encumberedBalanceOf[a]);
+    }
+
+    function _encumber(address owner, address taker, uint amount) private {
+        require(availableBalanceOf(owner) >= amount, "insufficient balance");
+        encumbrances[owner][taker] += amount;
+        encumberedBalanceOf[owner] += amount;
+        emit Encumber(owner, taker, amount);
+    }
+
+    function _release(address owner, address taker, uint amount) private {
+        if (encumbrances[owner][taker] < amount) {
+          amount = encumbrances[owner][taker];
+        }
+        encumbrances[owner][taker] -= amount;
+        encumberedBalanceOf[owner] -= amount;
+        emit Release(owner, taker, amount);
+    }
+
+    function transfer(address dst, uint amount) public override returns (bool) {
+        // check but dont spend encumbrance
+        require(availableBalanceOf(msg.sender) >= amount, "insufficient balance");
+        _transfer(msg.sender, dst, amount);
+        return true;
+    }
+
+    function transferFrom(address src, address dst, uint amount) public override returns (bool) {
+        uint encumberedToTaker = encumbrances[src][msg.sender];
+        bool exceedsEncumbrance = amount > encumberedToTaker;
+        if (exceedsEncumbrance)  {
+            uint excessAmount = amount - encumberedToTaker;
+
+            // check that enough enencumbered tokens exist to spend from allowance
+           require(availableBalanceOf(src) >= excessAmount, "insufficient balance");
+
+           // Exceeds Encumbrance , so spend all of it
+            _spendEncumbrance(src, msg.sender, encumberedToTaker);
+
+            _spendAllowance(src, dst, excessAmount);
+        } else {
+            _spendEncumbrance(src, msg.sender, amount);
+        }
+
+        _transfer(src, dst, amount);
+        return true;
+    }
+
+    function _spendEncumbrance(address owner, address taker, uint256 amount) internal virtual {
+        uint256 currentEncumbrance = encumbrances[owner][taker];
+        require(currentEncumbrance >= amount, "insufficient encumbrance");
+        uint newEncumbrance = currentEncumbrance - amount;
+        encumbrances[owner][taker] = newEncumbrance;
+        encumberedBalanceOf[owner] -= amount;
+    }
+}
+```
+
+
+## Security Considerations
+
+Parties relying on `balanceOf` to determine the amount of tokens available for transfer should instead rely on `balanceOf(account) - encumberedBalance(account)`, or, if implemented, `availableBalanceOf(account)`.
+
+The property that encumbered balances are always backed by a token balance can be accomplished in a straightforward manner by altering `transfer` and `transferFrom` to block . If there are other functions that can alter user balances, such as a rebasing token or an admin burn function, additional guards must be added by the implementer to likewise ensure those functions prevent reducing `balanceOf(account)` below `encumberedBalanceOf(account)` for any given account.
+Needs discussion.
```

Comment:
```suggestion
<!-- Needs discussion. -->
```

If you use HTML comments, the linter will make sure you remove them before moving to Review.